### PR TITLE
Fix long path init/re-init

### PR DESCRIPTION
**Pull request by nehaljwani**
Fixes https://github.com/conda/conda/issues/7952

### CODE REVIEW

**Comment by nehaljwani** [comment]
Fixes https://github.com/conda/conda/issues/7952

File: conda/core/initialize.py [hunk 1]
```diff
@@ -1187,13 +1189,13 @@ def init_long_path(target_path):
     # win10, build 14352 was the first preview release that supported this
     if int(win_ver) >= 10 and int(win_rev) >= 14352:
         prev_value, value_type = _read_windows_registry(target_path)
-        if prev_value != "1":
+        if prev_value != 1:
```
Comment (kalefranz) [inline conda/core/initialize.py]: Given the ambiguity here, whey not cast it just to be safe? `int(prev_value)`

Comment (kalefranz) [reply conda/core/initialize.py]: Actually, the safer choice is probably `if str(prev_value) != "1":` in case `prev_value` is `None`

Comment (nehaljwani) [reply conda/core/initialize.py]: Hmm... reading a registry value of type REG_DWORD should always return an int. But I understand that we wouldn't want a traceback in bad scenarios anyway.

**Pull request merged**